src/core/sprites/Sprite.js

```diff
@@ -407,20 +407,27 @@ Sprite.prototype.containsPoint = function( point )
 
 
 /**
- * Destroys this sprite and optionally its texture
+ * Destroys this sprite and optionally its texture and children
  *
- * @param [destroyTexture=false] {boolean} Should it destroy the current texture of the sprite as well
- * @param [destroyBaseTexture=false] {boolean} Should it destroy the base texture of the sprite as well
+ * @param [opts] {object} Options parameter
+ * @param [opts.children=false] {boolean} if set to true, all the children will have their destroy
+ *      method called as well. 'opts' will be passed on to those calls.
+ * @param [opts.texture=false] {boolean} Should it destroy the current texture of the sprite as well
+ * @param [opts.baseTexture=false] {boolean} Should it destroy the base texture of the sprite as well
  */
-Sprite.prototype.destroy = function (destroyTexture, destroyBaseTexture)
+Sprite.prototype.destroy = function (opts)
 {
-    Container.prototype.destroy.call(this);
+    opts = opts || {};
+    var texture     = opts.hasOwnProperty('texture')     ? opts.texture     : false;
+    var baseTexture = opts.hasOwnProperty('baseTexture') ? opts.baseTexture : false;
+
```

Comment:
Similar feedback to above, if the default is false then just do the check or use `!!` to normalize to a boolean if you *have* to. In this case you don't, just do the check normally and skip this extra work.